students/rhamersky/Lesson_10/mailroom_part6.py

```diff
@@ -0,0 +1,392 @@
+str_valediction = ""  # --> Email valediction
+
+
+# -----Process Section-----
+# This class creates an donor object.
+class Donor:
+    initial_donation = 1  # --> Field is used as the initial donation is constant with all donor objects.
+
+    def __init__(self, donor, donation):  # --> Constructs attributes donor and donation.
+        self.donor = donor
+        self.donation = donation
+
+########### END OF DONOR OBJECT CLASS###############
+
+# Methods in the Menu class will execute code based off of the user's menu choice.
+
+
+class Menu:
+    @staticmethod
+    def menu_options():
+        '''
+        Try and except block used on User's option choice to check for integers.
+        :return: Returns int data (1-4) which is used to determine what menu option will be executed.
+        '''
+        while True:
+            try:
+                int_choice = int(input(str_name + ", please select an option above. "))
+            except ValueError:
+                print(str_name + ", not a valid option. \n")
+            # Verify that the user selected an option 1 through option 4.
+            else:
+                if 1 <= int_choice <= 5:
+                    break
+                else:
+                    print("Integer selected is not a valid option. "
+                          "Please, enter an integer and corresponds to the options given. \n")
+        return int_choice
+
+    @staticmethod
+    def send_thankyou():
+        '''
+        This method will  allow new donors to be added, store donations, and print a thank you
+        statement to send to the donor by mail.
+        :return:
+        '''
+        donor_name = input("Please, enter full name of donor. ").title()
+        global dic_donors
+
+        # Gathers donation information from the user and prints a thank-you message for the donor.
+        while True:
+            # Appends the donor list.
+            if donor_name == "List":
+                # --> Verifies that there is at least a donor is stored as a list row in the table tuple.
+                if len(dic_donors) > 0:
+                    Data.print_donor_names()  # --> Prints a list of donors
+                    break
+                else:
+                    print(
+                        "\n" + "There are no donors in list.")  # --> Let's the user know that a donor list does not exist.
+                    break
+
+            try:
+                flo_donation_amount = float(input(str_name + ", please enter donation amount. "))
+
+                # Updates donor tuple
+                new_donor = Donor(donor_name,flo_donation_amount)
+                dic_donors = Data.update_donor(new_donor.donor, new_donor.donation, new_donor.initial_donation)
+
+                # String formatting for a brief thank-you to donor.
+                thankyou_message = "\n"+"{}, thank you for donating ${:,.2f}.".format(donor_name, flo_donation_amount)
+                print(thankyou_message)
+                break
```

Comment:
You should include as little code as possible in a `try` block and only the code that might raise the expected exception, so you don't accidentally catch unanticipated errors which can lead to some difficult debugging